openassessment/assessment/test/test_peer.py

```diff
@@ -1755,15 +1754,86 @@ def test_flexible_peer_grading__zero_graded(self):
         self.assertIsNone(peer_api.get_score(submission['uuid'], requirements))
 
     @staticmethod
-    def _create_student_and_submission(student, answer, date=None):
+    def _create_student_and_submission(student, answer, date=None, steps=None):
         """ Creats a student and submission for tests. """
         new_student_item = STUDENT_ITEM.copy()
         new_student_item["student_id"] = student
         submission = sub_api.create_submission(new_student_item, answer, date)
         peer_api.on_start(submission["uuid"])
-        workflow_api.create_workflow(submission["uuid"], STEPS)
+        workflow_api.create_workflow(submission["uuid"], steps or STEPS)
         return submission, new_student_item
 
+    def _assert_assessment_workflow_status(self, uuid, expected_status, step_requirements):
+        workflow = workflow_api.get_workflow_for_submission(uuid, step_requirements)
+        self.assertEqual(workflow['status'], expected_status)
+
+    def test_get_waiting_step_details__peer_item_created_not_assessed(self):
+        step_requirements = {'peer': {'must_grade': 1, 'must_be_graded_by': 2}}
+
+        # a target student and submission, and some other students and submissions
+        target_learner_sub, target_learner = self._create_student_and_submission(
+            'TargetLearner',
+            'TargetLearner submission',
+            steps=['peer']
+        )
+        other_learner_submissions = [
+            self._create_student_and_submission(f'Student{i}', f'Student{i} submission', steps=['peer'])
+            for i in range(5)
+        ]
+
+        # Have the target student submit her required assessment, they should now be waiting.
+        peer_api.get_submission_to_assess(target_learner_sub['uuid'], target_learner['student_id'])
+        peer_api.create_assessment(
+            target_learner_sub['uuid'],
+            target_learner['student_id'],
+            ASSESSMENT_DICT['options_selected'],
+            ASSESSMENT_DICT['criterion_feedback'],
+            ASSESSMENT_DICT['overall_feedback'],
+            RUBRIC_DICT,
+            step_requirements['peer']['must_be_graded_by']
+        )
+        self._assert_assessment_workflow_status(target_learner_sub['uuid'], 'waiting', step_requirements)
+
+        # Call get_submission_to_assess once more so that target_learner has an open incomplete peer assessment
+        peer_api.get_submission_to_assess(target_learner_sub['uuid'], target_learner['student_id'])
+
+        # Call get_submission_to_assess so all five learners in other_learner_submissions are
+        # currently assessing target_learner
+        for sub, student in other_learner_submissions:
+            chosen_submission = peer_api.get_submission_to_assess(sub['uuid'], student['student_id'])
+            self.assertIsNotNone(chosen_submission)
+            self.assertEqual(chosen_submission['uuid'], target_learner_sub['uuid'])
```

Comment:
This currently will always be true, but the rest of the test depends on it, so i'm asserting here just in case something changes in the future